src/app/app.module.ts

```diff
@@ -56,24 +47,25 @@ export class XhrInterceptor implements HttpInterceptor {
 }
 
 @Injectable()
-export class HttpXsrfInterceptor implements HttpInterceptor {
+export class AuthHttpInterceptor implements HttpInterceptor {
+    storage: Storage;
 
-    constructor(private tokenExtractor: HttpXsrfTokenExtractor, private injector: Injector) {
+    constructor(storage: Storage) {
+        this.storage = (storage === undefined || storage === null) ? sessionStorage : storage;
```

Comment:
this authentication mechanism is temporary anyway, so it would be easier to work on this when we have localStorage. That will allow us opening in multiple tabs without re-login. This will be replaced anyway by Idam, and would need to be handled differently also assuring proper logout function.